typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -56,55 +57,92 @@ const walletBalance = new Gauge({
+    .demandOption('e')
 
-/** How long we should wait for a message to be received in milliseconds. 10 min by default. */
-const MESSAGE_RECEIPT_TIMEOUT =
-  parseInt(process.env['KATHY_MESSAGE_RECEIPT_TIMEOUT'] as string) ||
-  10 * 60 * 1000;
+    .coerce('context', assertContext)
+    .demandOption('context')
 
-/** The maximum number of messages we will allow to get queued up if we are sending too slowly. */
-const MAX_MESSAGES_ALLOWED_TO_SEND = 5;
+    .boolean('cycle-once')
+    .describe(
+      'cycle-once',
+      'If true, will cycle through all chain pairs once as quick as possible',
+    )
+    .default('cycle-once', false)
+
+    .number('full-cycle-time')
+    .describe(
+      'full-cycle-time',
+      'How long it should take to go through all the message pairings in milliseconds. Ignored if --cycle-once is true. Defaults to 6 hours.',
+    )
+    .default('full-cycle-time', 1000 * 60 * 60 * 6) // 6 hrs
+
+    .number('message-send-timeout')
+    .describe(
+      'message-send-timeout',
+      'How long to wait for a message to be sent in milliseconds. Defaults to 10 min.',
+    )
+    .default('message-send-timeout', 10 * 60 * 1000) // 10 min
+
+    .number('message-receipt-timeout')
+    .describe(
+      'message-receipt-timeout',
+      'How long to wait for a message to be received on the destination in milliseconds. Defaults to 10 min.',
+    )
+    .default('message-receipt-timeout', 10 * 60 * 1000) // 10 min
+
+    .string('chains-to-skip')
+    .array('chains-to-skip')
+    .describe('chains-to-skip', 'Chains to skip sending from or sending to.')
+    .default('chains-to-skip', [])
+    .demandOption('chains-to-skip')
+    .coerce('chains-to-skip', (chainStrs: string[]) =>
+      chainStrs.map((chainStr: string) => assertChain(chainStr)),
+    ).argv;
+}
 
 async function main() {
+  const {
+    e: environment,
+    context,
+    chainsToSkip,
+    cycleOnce,
+    fullCycleTime,
+    messageSendTimeout,
+    messageReceiptTimeout,
+  } = await getKathyArgs();
+
   startMetricsServer(metricsRegister);
-  const environment = await getEnvironment();
   debug('Starting up', { environment });
+
   const coreConfig = getCoreEnvironmentConfig(environment);
-  const context = await getContext();
   const app = await getApp(coreConfig, context, KEY_ROLE_ENUM.Kathy);
   const gasCalculator = InterchainGasCalculator.fromEnvironment(
     environment,
     app.multiProvider as any,
   );
-  const chains = app.chains() as Chains[];
-  const skip = process.env.CHAINS_TO_SKIP?.split(',').filter(
-    (skipChain) => skipChain.length > 0,
-  );
+  const appChains = app.chains();
 
-  const invalidChains = skip?.filter(
-    (skipChain: any) => !chains.includes(skipChain),
-  );
-  if (invalidChains && invalidChains.length > 0) {
-    throw new Error(`Invalid chains to skip ${invalidChains}`);
+  // Ensure the specified chains to skip are actually valid for the app
+  for (const chainToSkip of chainsToSkip ?? []) {
```

Comment:
I think it would be better to default `chainsToSkip` in the yargs config to `[]`.

Reply:
I may be doing something wrong but for some reason yargs/typescript things that `chainsToSkip` can be undefined even though I'm setting the default to `[]` and i'm demanding the option. This is my workaround, adding a comment to indicate this. I agree this is weird

Reply:
I would instead suggest using `!` when you extract it from yargs if you know you mandated it.

Reply:
lmao what's wrong with me

Reply:
forgot that would work there